src/content/language/macros.md

```diff
@@ -0,0 +1,300 @@
+---
+title: Macros (experimental)
+description: Learn about the experimental macros feature as it develops.
+---
+
+[The Dart macro system][spec] is a major new language feature
+***currently under development*** which adds support for
+[static meta-programming][motivation] to the Dart language.
+
+A Dart macro is a user-defineable piece of code that takes in other code as parameters
+and operates on it in real-time to create, modify, or add declarations.
+
+You can think about the macro system in two parts: using macros and writing macros.
+This page covers each (at a high level, as ***the feature is still in preview***)
+in the following sections:
+
+- [**The `JsonCodable` macro**](#the-jsoncodable-macro):
+A ready-made macro you can try out today (behind an experiment flag)
+that offers a seamless solution to the
+common issue of tedious JSON serialization in Dart.
+
+- [**The macros feature in general**](#the-macros-language-feature):
+Why we're adding macros to Dart, motivating use cases,
+benefits over existing code gen solutions,
+and a cursory overview of how writing macros will work in the future once
+the feature is complete.
+
+[spec]: {{site.repo.dart.lang}}/blob/main/working/macros/feature-specification.md
+[motivation]: {{site.repo.dart.lang}}/language/blob/main/working/macros/motivation.md
+
+## The `JsonCodable` macro
+
+:::important
+The `JsonCodable` macro is not stable and currently behind an [experiment flag][].
+It only works with the [dev channel][channel] releases of Dart 3.5.0 and later.
```

Comment:
Just realized that Flutter no longer has a "dev channel". So I think we need to change `with the [dev channel][channel] releases of Dart 3.5.0 and later` to:

```
with Dart 3.5.0-152 or later. 
This is available from the [Dart dev channel][channel] 
or from the [Flutter master channel][flutter-channel].

[flutter-channel]: https://docs.flutter.dev/release/upgrade#other-channels
```